src/components/Player.jsx

```diff
@@ -36,18 +44,45 @@ function Player() {
     ));
 
     const rollDice = () => {
-        let allDice = [...playerState.dice];
-        allDice.forEach(die => {
-            if (die.locked === false) {
-                die.side = getRandomNumber()
+        //TODO: Restrict player from rolling if all dice are locked
+
+        //check if the player still has turns left
+        if (turnCount < 14) {
+            //check to make sure player still has at least one roll left during current turn
+            if (rollCount < 4) {
+                let allDice = [...playerState.dice];
+                allDice.forEach(die => {
+                    //only roll unlocked dice
+                    if (die.locked === false) {
+                        die.side = getRandomNumber()
+                    }
+                })
+                setPlayerState({ dice: allDice })
```

Comment:
This could be stylistic more than anything but I think I would have written this section as:
```suggestion
                setPlayerState({dice: 
                  [...playerState.dice].map(d => ({
                      ...d,
                      side: die.locked ? die.side : getRandomNumber()
                    })
                  );    
                });
```